File: citySimNGView/viewmodel/ViewModel.py
from viewmodel.GameMenuViewModel import GameMenuViewModel
from viewmodel.LoaderViewModel import LoaderViewModel
from viewmodel.MainMenuViewModel import MainMenuViewModel
from CreatorViewModel import  CreatorViewModel
from MapViewModel import MapViewModel
from TutorialViewModel import TutorialViewModel


class ViewModel(object):
    def __init__(self, viewSetter):
        self.viewSetter = viewSetter
        self.mainMenuViewModel = MainMenuViewModel(viewSetter)
        self.creatorViewModel = CreatorViewModel(viewSetter)
        self.gameMenuViewModel = GameMenuViewModel(viewSetter)
        self.loaderViewModel = LoaderViewModel(viewSetter)
        self.mapViewModel = MapViewModel(viewSetter)
        self.tutorialViewModel = TutorialViewModel(viewSetter)

    def getGameMenuViewModel(self):
        return self.gameMenuViewModel

    def getCreatorViewModel(self):
        return self.creatorViewModel

    def getMainMenuViewModel(self):
        return self.mainMenuViewModel

    def getLoaderViewModel(self):
        return self.loaderViewModel

    def getMapViewModel(self):
        return self.mapViewModel

    def getTutorialViewModel(self):
        return self.tutorialViewModel

    class Java:
        implements = ["py4jmediator.ViewModel"]

File: controllerModules/SkeletonModule/src/py4jmediator/ViewModel.java
package py4jmediator;

import entities.Building;
import entities.Dweller;
import entities.Resource;
import org.json.JSONObject;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

import org.json.JSONObject;

public interface ViewModel {

	public MainMenuViewModel getMainMenuViewModel();
	public CreatorViewModel getCreatorViewModel();
	public GameMenuViewModel getGameMenuViewModel();
	public LoaderViewModel getLoaderViewModel ();
	public MapViewModel getMapViewModel();
	public TutorialViewModel getTutorialViewModel();

	public interface MapViewModel{
		public void displayMap();
		public void init(List<Resource> resources, List<Building> domesticBuildings,
						 List<Building> industrialBuildings, List<Dweller> dwellers,
						 String texture_one, String texture_two, String panelTexture, String mp3,
						 Map<String, Integer> initialResourcesValues,
						 Map<String, Integer> initialResourcesIncomes,
						 Map<String, Integer> actualResourcesConsumption,
						 Map<String, Integer> resourcesBalance, int availableDwellers);
		public void updateValuesForCycle(Map<String, Integer> actualResourcesValues,
										 Map<String, Integer> actualResourcesIncomes,
										 Map<String, Integer> actualResourcesConsumption,
										 Map<String, Integer> resourcesBalance,
										 Integer neededDwellers,
										 Integer availableDwellers);
		public void sendTutorialHints(String hints);
		public void resumeGame();
	}

	public interface MainMenuViewModel{
		public void displayMainMenu();
	}

	public interface GameMenuViewModel{
		public void displayGameMenu();
		public void animateCurrentPrices(HashMap<String, Integer> currentPrices);
	}

	public interface CreatorViewModel{
		public void displayCreator();
		public void displayDependenciesGraph(JSONObject jsonGraph);
		public void displayMsg(String errorMsg);
	}

	public interface LoaderViewModel{
		public void displayLoader();
		public void displayDependenciesGraph(JSONObject graphDesc);
		public void displayPossibleDependenciesSets(Set<String> possibleSets);
	}

	public interface TutorialViewModel {
		public void displayTutorial();
		public void displayTutorialPage(JSONObject jsonPage);
		public void displayDependenciesGraph(JSONObject jsonGraph);
		public void fetchTutorialIndex(String[] index);
		public void fetchNodes(List<String> buildingsList, 
		List<String> resourcesList, List<String> dwellersList );
	}
}